src/atc/etl/loader.py

```diff
@@ -1,24 +1,31 @@
 from abc import abstractmethod
-from typing import List
 
 from pyspark.sql import DataFrame
 
+from atc.etl.types import dataset_group, EtlBase
 
-class Loader:
-    @abstractmethod
-    def save(self, df: DataFrame) -> DataFrame:
-        return df
 
+class Loader(EtlBase):
```

Comment:
I agree that a `Loader` is by design a data sink. The `DataFrame` return type was something I added previously for `Loader` implementations that target CosmosDb, where the document ID and partition key are required by the data store but not relevant on the data record. Returning the `DataFrame` with the new columns made sense at the time. In a way it makes more sense that its the `Transformer` that appends these columns but then the `Transformer` is now responsible for data storage specific concerns, and not purely data transformation

Reply:
I am familiar with the use case, and it is fine to both add it in the loader or in a transformer. The real reason to not return it here is to make it clear that in te case of loader->loader->loader, the loaders will be called with the same dataframe and not sequentially enriched dataframes.